core/embed/rust/src/ui/model_tr/bootloader/mod.rs

```diff
@@ -295,31 +256,105 @@ extern "C" fn screen_install_success(
     show(&mut frame, false);
 }
 
-#[no_mangle]
-extern "C" fn screen_welcome() {
-    let mut frame = Welcome::new();
-    show(&mut frame, false);
-}
-
-#[no_mangle]
-extern "C" fn bld_continue_label(bg_color: cty::uint16_t) {
+fn bld_continue_label(bg_color: Color) {
     display::text_center(
         Point::new(constant::WIDTH / 2, HEIGHT - 2),
         "CONTINUE",
         Font::NORMAL,
         WHITE,
-        Color::from_u16(bg_color),
+        bg_color,
     );
     ICON_ARM_LEFT.draw(
         Point::new(constant::WIDTH / 2 - 36, HEIGHT - 6),
         Alignment2D::TOP_LEFT,
         WHITE,
-        Color::from_u16(bg_color),
+        bg_color,
     );
     ICON_ARM_RIGHT.draw(
         Point::new(constant::WIDTH / 2 + 25, HEIGHT - 6),
         Alignment2D::TOP_LEFT,
         WHITE,
-        Color::from_u16(bg_color),
+        bg_color,
     );
 }
+
+impl UIFeaturesBootloader for ModelTRFeatures {
+    fn screen_welcome() {
+        let mut frame = Welcome::new();
+        show(&mut frame, true);
+    }
+
+    fn bld_continue_label(bg_color: Color) {
+        bld_continue_label(bg_color);
```

Comment:
any reason not to inline the functions into the trait implementation?

Reply:
i was just afraid of merging with https://github.com/trezor/trezor-firmware/pull/3649 , which also edits this file heavily. But it can be handled i guess if you prefer it

Reply:
i would expect git to handle this fine, and if not, 100% sure vscode merge editor will deal with it. please do it that way.

Reply:
fixed https://github.com/trezor/trezor-firmware/commit/6625666b53940c96d24f0384a31cf434206707fd